manifests/record.pp

```diff
@@ -0,0 +1,56 @@
+# Define new DNS record
+define dns::record (
+  $zone     = undef,
+  $label    = undef,
+  $target   = undef,
+  $type     = undef,
+  $priority = undef,
+  $weight   = undef,
+  $port     = undef,
+  $comment  = ' ',
+) {
+
+  $dnstype = upcase($type)
+
+  if $target == undef {
+    fail("No target specified for record ${title}")
+  }
+
+  if $dnstype == undef {
```

Comment:
I think I can even totally drop the check for empty $dnstype, as it's caught by the default of the if statement starting line 34 (which I should probably be replacing with a case statement by now :wink: )